internal/indexer/postgres/affectedmanifest.go

```diff
@@ -0,0 +1,269 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/rs/zerolog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/pkg/omnimatcher"
+)
+
+var (
+	// ErrNotIndexed indicates the vulnerability being queried has a dist or repo not
+	// indexed into the database.
+	ErrNotIndexed = fmt.Errorf("vulnerability containers data not indexed by any scannners")
+)
+
+// affectedManifests finds the manifests digests which are affected by the provided vulnerability.
+//
+// an exhaustive search for all indexed packages of the same name as the vulnerability is performed.
+//
+// the list of packages is filtered down to only the affected set.
+//
+// the manifest index is then queried to resolve a list of manifest hashes containing the affected
+// artifacts.
+func affectedManifests(ctx context.Context, pool *pgxpool.Pool, v claircore.Vulnerability) ([]claircore.Digest, error) {
+	const (
+		selectPackages = `
+		SELECT id, name, version, kind
+		FROM package
+		WHERE name = $1;
+		`
+		selectAffected = `
+		SELECT manifest_hash
+		FROM manifest_index
+		WHERE package_id = $1
+		  AND (
+			CASE
+				WHEN $2::bigint IS NULL THEN
+					dist_id IS NULL
+				ELSE
+					dist_id = $2
+				END
+			)
+		  AND (
+			CASE
+				WHEN $3::bigint IS NULL THEN
+					repo_id IS NULL
+				ELSE
+					repo_id = $3
+				END
+			);
+		`
+	)
+	log := zerolog.Ctx(ctx).
+		With().
+		Str("component", "internal/indexer/postgres/affectedManifests").
+		Logger()
+
+	// confirm the incoming vuln can be
+	// resolved into a prototype index record
+	pr, err := protoRecord(ctx, pool, v)
+	switch {
+	case err == nil:
+		// break out
+	case errors.Is(err, ErrNotIndexed):
+		log.Warn().Str("vulnID", v.ID).Msg("vuln is comprised of repo or distro not indexed by any scanner yet")
+		return nil, nil
+	default:
+		return nil, err
+	}
+
+	// collect all packages which may be affected
+	// by the vulnerability in question.
+	pkgsToFilter := []claircore.Package{}
+	rows, err := pool.Query(ctx, selectPackages, v.Package.Name)
+	if err != nil {
+		rows.Close()
```

Comment:
```suggestion
	if rows != nil {
		defer rows.Close()
	}
	if err != nil {
```

This one isn't done in a loop or transaction, so I think the close can just be deferred.

Reply:
yeah I think you are right.